src/pages/Company/CompanyTemplate.tsx

```diff
@@ -25,82 +26,106 @@ export const CompanyTemplate: React.FC<ICompanyTemplateProps> = ({
   notifications
 }) => {
   const windowSize = useWindowSize();
+  const [ students, setStudents ] = useState<IStudentDetailedDataProps[]>([]);
+  const [ studentsInBookmarks, setStudentsInBookmarks ] = useState<number[]>([]);
+
+  useEffect(() => {
+    const fetchStudents = async () => {
+      const response = await fetch('https://69dd40e2-488f-4731-b0e0-b4671a6138ae.mock.pstmn.io/students');
+      const studentsArray = await response.json();
+
+      setStudents(Object.values(studentsArray));
+    };
+
+    const fetchStudentsInBookmarks = async () => {
+      const response = await fetch('https://69dd40e2-488f-4731-b0e0-b4671a6138ae.mock.pstmn.io/bookmarked-students');
+      const studentsIdArray = await response.json();
+
+      setStudentsInBookmarks(studentsIdArray.studentsInBookmarks);
+    };
+
+    Promise.all([fetchStudents(), fetchStudentsInBookmarks()]).catch(console.error);
```

Comment:
Move core logic out from components itself.
The component should just getting some data (from your core hook for example).
```
const { data } = useGetStudents();
```

Hook implementation could be in some `core` directory.

Btw here is example of directory's structure:
```
core/hooks:
    index.ts
    [Hook's folder name]:
        index.ts
        index.test.ts // optional
        HookName.ts
        [transform]:
            index.ts
            transformDataFromServer.ts
            anotherTransform.ts
```